src/main/java/com/urlparser/entity/LinksFromUrl.java

```diff
@@ -0,0 +1,53 @@
+package com.urlparser.entity;
+
+import com.urlparser.model.Links;
+
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.jsoup.select.Elements;
+import org.apache.log4j.LogManager;
+import org.apache.log4j.Logger;
+
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+
+public class LinksFromUrl implements Links {
+
+  private static Logger log = LogManager.getLogger(LinksFromUrl.class);
+  private final String url;
+
+  public LinksFromUrl(String url) {
+    this.url = url;
+  }
+
+  private static List<URL> links(String url) {
+    Document doc = null;
+    List<URL> listedLinks = new ArrayList<>();
+    try {
+      doc = Jsoup.connect(url).get();
+    } catch (IOException e) {
+      log.error("Error during connect to website", e);
+    }
+    Elements links = doc.select("a[href]");
```

Comment:
@undsett lets simplify this method. 
Lets accept as a constructor parameter `Links<String> origin` for this class and introduce `Links` implementation that operates with `String` encapsulating all the HTML parsing logic.